roachpb/data.go

```diff
@@ -602,32 +602,21 @@ func (t *Transaction) IsInitialized() bool {
 	return len(t.ID) > 0
 }
 
-// MakePriority generates a random priority value, biased by the
-// specified userPriority. If userPriority=100, the resulting
-// priority is 100x more likely to be probabilistically greater
-// than a similar invocation with userPriority=1.
+// MakePriority returns the transaction priority based on the
+// specified user priority. If 0, the user priority is set to
+// a random value between 0 and max int32.
 func MakePriority(userPriority int32) int32 {
-	// A currently undocumented feature allows an explicit priority to
-	// be set by specifying priority < 1. The explicit priority is
-	// simply -userPriority in this case. This is hacky, but currently
-	// used for unittesting. Perhaps this should be documented and allowed.
+	// A former undocumented feature allowed an explicit priority to be
+	// set by specifying priority < 1. The explicit priority is simply
+	// -userPriority in this case. This is used for unittesting, before
+	// the specified priority was taken at face value, so we maintain
+	// the old behavior.
```

Comment:
There aren't that many tests that use negative priorities, are there? I think it would be nice to clean those up and get rid of this special case.